doc/user/content/transform-data/patterns/partition-by.md

```diff
@@ -0,0 +1,145 @@
+
+Materialize lets you specify the ordering it will use to sort these runs of data internally. A well-chosen sort order can unlock optimizations like [filter pushdown](#filter-pushdown), which in turn can make queries and other operations more efficient.
+
+## Syntax
+
+The option `PARTITION BY <column list>` declares that a [materialized view](/sql/create-materialized-view/#with_options), [table](/sql/create-table/#with_options), or source table should be ordered by the listed columns. For example, a table that stores an append-only collection of events may look like:
+
+```mzsql
+CREATE TABLE events (created_at timestamptz, body jsonb)
+WITH (
+    PARTITION BY (created_at)
+);
+```
+
+When multiple columns are specified, rows are ordered lexicographically:
+`PARTITION BY (created_date, created_time)` would sort first by the created date, then order rows with equal created date by time. Durable collections without a `PARTITION BY` option might be stored in any order.
+
+{{< note >}}
+The `PARTITION BY` option has no impact on the order in which records are returned by queries. If you want to return results in a specific order, use an `ORDER BY` clause on your select statement.
+{{< /note >}}
+`ORDER BY` clause on your select statement.
+
+## Filter pushdown
+
+As mentioned above, Materialize stores durable collections as a set of parts. If it can prove that a particular part in a collection is not needed to answer a particular query, it can skip fetching it, saving a substantial amount of time and computation. For example:
+
+- If a materialized view has a temporal filter that preserves only the last two days of data, we'd like to filter out parts that contain data from a month ago.
+- If a select statement filters for only data from a particular country, we'd like to avoid fetching parts that only contain data from other countries.
+
+This optimization tends to be important for append-only timeseries datasets, but it can be useful for any dataset where most queries look at only a particular "range" of the data. If the query is a select statement, it can make that select statement more performant; if the query is an index or a materialized view, it can make it much faster to bootstrap.
+
+Materialize will always try to filter out parts it doesn't need for a particular query, but that filtering is usually only effective when similar rows are stored together. If you want to make sure that filter pushdown reliably kicks in for your query, you should:
+
+- Use a `PARTITION BY` clause on the relevant column to ensure that data with similar values for that column are stored close together.
+- Add a filter to your query that only returns true for a particular range of values in that column.
```

Comment:
I felt that the rewrite here was a bit unclear about the link between filters and the optimization, but I agree the first version was not quite right either. Tried a rewrite based on some of your ideas here in case any of that looks better?